wagon.py

```diff
@@ -0,0 +1,1015 @@
+                requirement_files, process.aggr_stderr))
+
+    wheel_command = _construct_wheel_command(
+        wheels_path,
+        excluded_packages,
+        wheel_args,
+        package=package)
+    process = run(wheel_command)
+    if not process.returncode == 0:
+        raise WagonError('Could not download wheels for: {0} ({1})'.format(
+            package, process.aggr_stderr))
+
+    wheels, excluded_wheels = _get_wheels(
+        wheels_path, excluded_packages, package)
+
+    return wheels, excluded_wheels
+
+
+def _get_wheel_for_package(wheels_path, package):
+    for wheel in os.listdir(wheels_path):
+        if wheel.startswith(package.replace('-', '_')):
+            return wheel
+
+
+def _construct_pip_command(package,
+                           wheels_path,
+                           virtualenv,
+                           requirements_file=None,
+                           upgrade=False,
+                           install_args=None):
+    pip_executable = _get_pip_path(virtualenv)
+
+    pip_command = [pip_executable, 'install']
+    if requirements_file:
+        pip_command.extend(['-r', requirements_file])
+    if install_args:
+        pip_command.append(install_args)
+    pip_command.append(package)
+    pip_command.extend(
+        ['--use-wheel', '--no-index', '--find-links', wheels_path])
+    # pre allows installing both prereleases and regular releases depending
+    # on the wheels provided.
+    pip_command.append('--pre')
+    if upgrade:
+        pip_command.append('--upgrade')
+
+    return ' '.join(pip_command)
+
+
+def install_package(package,
+                    wheels_path,
+                    virtualenv=None,
+                    requirements_file=None,
+                    upgrade=False,
+                    install_args=None):
+    """This will install a Python package.
+
+    Can specify a specific version.
+    Can specify a prerelease.
+    Can specify a virtualenv to install in.
+    Can specify a list of paths or urls to requirement txt files.
+    Can specify a local wheels_path to use for offline installation.
+    Can request an upgrade.
+    """
+    logger.info('Installing {0}...'.format(package))
+    if virtualenv and not os.path.isdir(virtualenv):
+        raise WagonError('Virtualenv {0} does not exist'.format(virtualenv))
+
+    pip_command = _construct_pip_command(
+        package,
+        wheels_path,
+        virtualenv,
+        requirements_file,
+        upgrade,
+        install_args)
+
+    if IS_VIRTUALENV and not virtualenv:
+        logger.info('Installing within current virtualenv: {0}...'.format(
+            IS_VIRTUALENV))
+
+    result = run(pip_command)
+    if not result.returncode == 0:
+        raise WagonError('Could not install package: {0} ({1})'.format(
+            package, result.aggr_stderr))
+
+
+def _get_downloaded_wheels(path):
+    return \
+        [filename for filename in os.listdir(path) if filename.endswith('whl')]
```

Comment:
How about using os.path.splitext for the file extension instead of testing for endswith?
IMHO os.path.splitext is more intention revealing, it is clear that you're specifically looking for the file extension. for instance let's say  I have a file that just ends with a 'whl' with no extension the condition will pass. which is a false positive result.